dask/dataframe/io/tests/test_sql.py

```diff
@@ -0,0 +1,134 @@
+from __future__ import (print_function, division, absolute_import,
+                        unicode_literals)
+
+import io
+import pytest
+
+from dask.dataframe.io.sql import read_sql_table
+from dask.utils import tmpfile
+from dask.dataframe.utils import assert_eq
+pd = pytest.importorskip('pandas')
+dd = pytest.importorskip('dask.dataframe')
+pytest.importorskip('sqlalchemy')
+pytest.importorskip('sqlite3')
+
+
+data = """
+name,number,age,negish
+Alice,0,33,-5
+Bob,1,40,-3
+Chris,2,22,3
+Dora,3,16,5
+Edith,4,53,0
+Francis,5,30,0
+Garreth,6,20,0
+"""
+
+df = pd.read_csv(io.StringIO(data), index_col='number')
+
+
+@pytest.yield_fixture
+def db():
+    with tmpfile() as f:
+        uri = 'sqlite:///%s' % f
+        df.to_sql('test', uri, index=True, if_exists='replace')
+        yield uri
+
+
+def test_simple(db):
+    # single chunk
+    data = read_sql_table('test', db, npartitions=2, index_col='number'
+                          ).compute()
+    assert (data.name == df.name).all()
+    assert data.index.name == 'number'
+    assert_eq(data, df)
+
+
+def test_npartitions(db):
+    data = read_sql_table('test', db, columns=list(df.columns), npartitions=2,
+                          index_col='number')
+    assert len(data.divisions) == 3
+    assert (data.name.compute() == df.name).all()
+    data = read_sql_table('test', db, columns=['name'], npartitions=6,
+                          index_col="number").compute()
```

Comment:
Remove the compute call here.  We want to let assert_eq call compute itself.  I would probably be a good idea for you to look through the [assert_eq code](https://github.com/dask/dask/blob/master/dask/dataframe/utils.py#L494) at some point.